2/2.js

```diff
@@ -0,0 +1,211 @@
+        runtime: "153 min",
+        genre: ["Action", "Adventure", "Family"],
+        director: "David Yates",
+        writer: "Steve Kloves, J.K. Rowling",
+        actors: ["Daniel Radcliffe", "Emma Watson", "Rupert Grint"],
+        plot: "As Harry Potter begins his sixth year at Hogwarts, he discovers an old book marked as 'the property of the Half-Blood Prince' and begins to learn more about Lord Voldemort\'s dark past.",
+        country: "United Kingdom",
+        poster: "https://m.media-amazon.com/images/M/MV5BNzU3NDg4NTAyNV5BMl5BanBnXkFtZTcwOTg2ODg1Mg@@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 492245,
+        boxOffice: "$302,305,431",
+        production: "Heyday Films, Warner Bros.",
+    },
+    {
+        title: "Harry Potter and the Sorcerer's Stone",
+        year: 2001,
+        released: "16 Nov 2001",
+        runtime: "152 min",
+        genre: ["Adventure", "Family", "Fantasy"],
+        director: "Chris Columbus",
+        writer: "J.K. Rowling, Steve Kloves",
+        actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+        plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+        country: "United Kingdom, United States",
+        poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 684604,
+        boxOffice: "$318,087,620",
+        production: "1492 Pictures, Heyday Films, Warner Brothers",
+    }
+]
+
+
+/*C приведенным ниже массивом фильмов сделать следующий задачи, используя *
+*map/reduce вместо for, forEach**: 
+
+1. Собрать в массив все жанры фильмов (без повторения)
+*/
+
+let allGenre = data.reduce((newArr, item) => {
+
+
+    let genre;
+    genre = item.genre;
+    newArr.push(genre);
+    return newArr;
+}, [])
+let allArrGenre = allGenre.join().split(',')
+
+let uniqGenre = allArrGenre.filter((item, index) => allArrGenre.indexOf(item) === index);
+
+console.log(uniqGenre)
+
+
+//2. Собрать в массив всех актеров всех фильмов (без повторения)
+
+let allActors = data.map(item => {
+    let actors = item.actors;
+    return actors;
+})
+
+let allArrActors = allActors.join().replaceAll(' ', '').split(',')
+
+let uniqActors = allArrActors.filter((item, index) => allArrActors.indexOf(item) === index);
+
+console.log(uniqActors)
+
+
+//3. Отсортировать фильмы по рейтингу по убыванию
+
+let listRaiting = data.map(item => {
+
+    let title = item.title;
+    let rating = item.imdbRating;
+
+    let film = {
+        name: title,
+        rating: rating
+    }
+
+    return film;
+})
+
+listRaiting.sort((a, b) => b.rating - a.rating);
+console.log(listRaiting);
+
+
+
+//4. Создать новый массив, где объекты фильмов будут состоять из следующих полей: 
+//`id, title, released, plot`
+
+let shortData = data.map(item => {
```

Comment:
ok)
but it can be more simple

const shortData = data.map(({ id, title, released, plot }) =>{ id, title, released, plot });